### PR TITLE
Improved Logging for skipped resources.

### PR DESCRIPTION
**Pull request by AdityaKBhadragond14**
**Feature Description:**

- Added a new Table in logs with the following structure to know more detailed info regarding the skipped resources during fhir to omop conversion:

              | Step                                     | Skip Reason                              | Skipped Count |
              -------------------------------------------------------------------------------------------------------
              | stepProcessPatients                      | No Birth Date Found                      | 59           |
              | stepProcessConditions                    | No referenced person found               | 17           |
              |                                          | No startdate found                       | 2            |
              |                                          | No source code found                     | 154          |
              |                                          | No Diagnostic Confidence found for the resource | 27           |
              | stepProcessObservations                  | Status is not acceptable                 | 200          |
              |                                          | No Matching Encounter found for the resource | 29           |
              | stepProcessProcedures                    | No startdate found                       | 25           |
              |                                          | No source code found                     | 36           |
              |                                          | No Matching Encounter found for the resource | 83           |
              |                                          | The status is not acceptable             | 56           |
              | stepProcessImmunizations                 | No referenced person found               | 162          |
              |                                          | No startdate found                       | 4            |
              |                                          | No source code found                     | 34           |

Code Changes:

- In the `MapperMetrics` class added setters for the counts for each reasons. Each setter has a **name** and a **description**.
- The name is used in the `FhirToOmopJobListener` class in `getCounters` method which gets all the counts.
- The `getCounters` method is used in the `logSkippedReasons` method which is used to log the table.
- The setters are used to increment the count in each Mapper class.

### CODE REVIEW

File: src/main/java/org/miracum/etl/fhirtoomop/TaskConfiguration.java
Comment (joiskash) [inline src/main/java/org/miracum/etl/fhirtoomop/TaskConfiguration.java:1]: Please undo all the white space changes. This file is very difficult to review.

**Review by joiskash — CHANGES_REQUESTED** [review]
Looks good. Just want to review TaskConfiguration easily

File: src/main/java/org/miracum/etl/fhirtoomop/TaskConfiguration.java
Comment (AdityaKBhadragond14) [reply src/main/java/org/miracum/etl/fhirtoomop/TaskConfiguration.java:1]: Okay

Comment (AdityaKBhadragond14) [reply src/main/java/org/miracum/etl/fhirtoomop/TaskConfiguration.java:1]: Done please re review.

**Review by nsabale7 — APPROVED** [review]

**Pull request merged**